### PR TITLE
TUNIC: Location groups for each area of the game

### PR DESCRIPTION
**Pull request by ScipioWright**
## What is this fixing or adding?
Creates location groups for each area of the game, so every location is now in a ~~second~~ location group. We had this originally, but removed it because it was jank. But now it's not jank.

Also very very slight optimization in the hint code to use `name = string.split(" - ", 1)[0]` instead of `name, _ = string.split`

And scope creeping it a bit to add another item group for the wand, because I keep typing !hint wand and it keeps not working. Also adding an item group "fire rod" for the magic wand, since someone hinted that and fuzzy matching decided they were hinting for ice rod, which is icebolt, which is wrong

## How was this tested?
Test gens, adding location groups to priority locations and seeing what the spoiler thinks the priority locations are.

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

File: worlds/tunic/__init__.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def extend_hint_information(self, hint_data: Dict[int, Dict[int, str]]):
                     name, connection = connection
                     # for LS entrances, we just want to give the portal name
                     if "(LS)" in name:
-                        name, _ = name.split(" (LS) ")
+                        name = name.split(" (LS) ")[0]
```
Comment (Berserker66) [inline worlds/tunic/__init__.py:253]:
```suggestion
                        name = name.split(" (LS) ", 1)[0]
```
if we're optimizing anyway, may as well go all the way.

**Review by silent-destroyer — APPROVED** [review]

**Review by hatkirby — APPROVED** [review]
Generated weighted-options.json and saw that the location groups do exist. LGTM.

File: worlds/tunic/locations.py [hunk 2]
```diff
@@ -325,6 +325,10 @@ class TunicLocationData(NamedTuple):
 
 location_name_groups: Dict[str, Set[str]] = {}
 for loc_name, loc_data in location_table.items():
+    loc_group_name = loc_name.split(" - ", 1)[0]
+    if loc_group_name not in location_name_groups.keys():
+        location_name_groups[loc_group_name] = set()
+    location_name_groups[loc_group_name].add(loc_name)
```
Comment (NewSoupVi) [inline worlds/tunic/locations.py:331]:
```
location_name_groups: Dict[str, Set[str]] = {}
for loc_name, loc_data in location_table.items():
    loc_group_name = loc_name.split(" - ", 1)[0]
    location_name_groups.setdefault(loc_group_name, set()).add(loc_name)
```

or, in my opinion even better

```
location_name_groups: Dict[str, Set[str]] = defaultdict(lambda: set())
for loc_name, loc_data in location_table.items():
    loc_group_name = loc_name.split(" - ", 1)[0]
    location_name_groups[loc_group_name).add(loc_name)
```

**Review by NewSoupVi — APPROVED** [review]
Looks good to me. I don't like the pattern chosen for a dict of sets, I would prefer `setdefault` or `defaultdict`, but it's not breaking.

File: worlds/tunic/locations.py (hunk 2 above)
Comment (ScipioWright) [reply worlds/tunic/locations.py]: Revised it to the first suggestion cause I understand that one better

**Pull request merged**